axlearn/common/config.py

```diff
@@ -371,7 +371,14 @@ def to_flat_dict(self, *, omit_default_values: Collection[Any]) -> Dict[str, Any
         result = {}
 
         def enter(key: str, val: Any, default_result: Optional[List]) -> Optional[List]:
-            if key and isinstance(val, ConfigBase):
+            if dataclasses.is_dataclass(val) and not isinstance(val, type):
```

Comment:
this is similar to the condition check of _default_enter_fn. https://github.com/apple/axlearn/blob/4c9ecd92c8d1a411934a239763a778c2f80825e8/axlearn/common/config.py#L487